### PR TITLE
Adds an optional RHACS install step

### PR DESCRIPTION
**Pull request by gavin-stackrox**
Installs RHACS after `post-install` if `rhacs_enable=true`.

Requirements for the ansible orchestration host (i.e. jump host) that may be additional are:
- helm
- kubernetes.core.helm (ansible collection)

Tested on RHEL 8.4 against an existing OS 4.7.18.

Tested with:
- ansible 2.9.23
- helm v3.5.0+6.el8
- kubernetes.core.helm 2.1.1

Tested with a `deploy-cluster.yml` play and had to set OPENSHIFT_WORKER_INSTANCE_TYPE=m5.xlarge.

### CODE REVIEW

**Comment by gavin-stackrox** [comment]
@mffiedler @jtaleric @mohit-sheth Is there anything you need from me on this PR? thanks.

**Comment by gavin-stackrox** [comment]
🤯 I feel like there was a link to CI at one point in this PR thread but now it has magically disappeared. 

i.e. https://jenkins.scalelab.redhat.com/job/scale-ci-deploy-CI/6/

which failed due to missing ansible collection. I expect the CI containers/hosts need:
`ansible-galaxy collection install kubernetes.core`
and probably:
`dnf -y install helm`

**Comment by gavin-stackrox** [comment]
> 🤯 I feel like there was a link to CI at one point in this PR thread but now it has magically disappeared.
> 
> i.e. https://jenkins.scalelab.redhat.com/job/scale-ci-deploy-CI/6/
> 
> which failed due to missing ansible collection. I expect the CI containers/hosts need:
> `ansible-galaxy collection install kubernetes.core`
> and probably:
> `dnf -y install helm`

@mohit-sheth thanks for your help getting this tested in CI. Do you have any advice re: the missing dependencies? thanks.

**Comment by gavin-stackrox** [comment]
@mffiedler @jtaleric @mohit-sheth do you have any input for me on this PR / the missing dependencies? thanks

File: OCP-4.X/roles/rhacs_install/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+#
+# Creates a RHACS install in the cluster
+#
+
+- set_fact:
```
Comment (jtaleric) [inline OCP-4.X/roles/rhacs_install/tasks/main.yml:6]: Curious why we are setting this fact versus retrieving it like have in the past?

**Comment by jtaleric** [comment]
> @mffiedler @jtaleric @mohit-sheth do you have any input for me on this PR / the missing dependencies? thanks

https://github.com/cloud-bulldozer/scale-ci-deploy/blob/master/OCP-4.X/roles/openshift-install/tasks/main.yml#L46-L66

Seems we could simply add he additional necessary packages here to be installed on the jumphost.

File: OCP-4.X/roles/rhacs_install/tasks/main.yml (hunk 1 above)
Comment (gavin-stackrox) [reply OCP-4.X/roles/rhacs_install/tasks/main.yml:6]: I did this to get a static value to use throughout this file.

**Comment by gavin-stackrox** [comment]
> > @mffiedler @jtaleric @mohit-sheth do you have any input for me on this PR / the missing dependencies? thanks
> 
> https://github.com/cloud-bulldozer/scale-ci-deploy/blob/master/OCP-4.X/roles/openshift-install/tasks/main.yml#L46-L66
> 
> Seems we could simply add he additional necessary packages here to be installed on the jumphost.
> 
> thoughts @mohit-sheth

I _think_ the ansible collection needs to be installed before ansible is run. (but I may be wrong there).

**Comment by mohit-sheth** [comment]
I have added ansible collections to out jump host that is used for the CI, you can add helm to the packages list andwe can rerun the CI @gavin-stackrox

**Comment by mohit-sheth** [comment]
```
14:29:12 [0;31mERROR! couldn't resolve module/action 'kubernetes.core.helm_repository'. This often indicates a misspelling, missing collection, or incorrect module path.[0m
14:29:12 [0;31m[0m
14:29:12 [0;31mThe error appears to be in '/var/jenkins/workspace/scale-ci-deploy-CI/OCP-4.X/roles/rhacs_install/tasks/main.yml': line 9, column 3, but may[0m
```
@gavin-stackrox

**Comment by gavin-stackrox** [comment]
hi @mohit-sheth, I'm going to guess that this has something to do with how the kubernetes.core collection was installed on the jump host. Or maybe there is a version difference with what was installed and what I tested. 
I'm using ansible & ansible-galaxy (2.9.23) and the kubernetes.core collection is 2.1.1. 
```
$ ls -l ~/.ansible/collections/ansible_collections/kubernetes/core/plugins/modules
total 152
-rw-r--r--. 1 gavin gavin  5259 Jun 30 16:19 helm_info.py
-rw-r--r--. 1 gavin gavin  4375 Jun 30 16:19 helm_plugin_info.py
-rw-r--r--. 1 gavin gavin  7304 Jun 30 16:19 helm_plugin.py
-rw-r--r--. 1 gavin gavin 20590 Jun 30 16:19 helm.py
-rw-r--r--. 1 gavin gavin  7286 Jun 30 16:19 helm_repository.py
-rw-r--r--. 1 gavin gavin  6373 Jun 30 16:19 helm_template.py
-rw-r--r--. 1 gavin gavin     0 Jun 30 16:19 __init__.py
-rw-r--r--. 1 gavin gavin  5819 Jun 30 16:19 k8s_cluster_info.py
-rw-r--r--. 1 gavin gavin  5392 Jun 30 16:19 k8s_exec.py
-rw-r--r--. 1 gavin gavin  5576 Jun 30 16:19 k8s_info.py
-rw-r--r--. 1 gavin gavin  8264 Jun 30 16:19 k8s_json_patch.py
-rw-r--r--. 1 gavin gavin  7418 Jun 30 16:19 k8s_log.py
-rw-r--r--. 1 gavin gavin 14009 Jun 30 16:19 k8s.py
-rw-r--r--. 1 gavin gavin  6781 Jun 30 16:19 k8s_rollback.py
-rw-r--r--. 1 gavin gavin 11323 Jun 30 16:19 k8s_scale.py
-rw-r--r--. 1 gavin gavin  7089 Jun 30 16:19 k8s_service.py
```

**Comment by mohit-sheth** [comment]
```
ansible [core 2.11.2]
kubernetes.core:2.1.1 was installed successfully

total 152
-rw-r--r--. 1 root root  5259 Jul  8 23:59 helm_info.py
-rw-r--r--. 1 root root  4375 Jul  8 23:59 helm_plugin_info.py
-rw-r--r--. 1 root root  7304 Jul  8 23:59 helm_plugin.py
-rw-r--r--. 1 root root 20590 Jul  8 23:59 helm.py
-rw-r--r--. 1 root root  7286 Jul  8 23:59 helm_repository.py
-rw-r--r--. 1 root root  6373 Jul  8 23:59 helm_template.py
-rw-r--r--. 1 root root     0 Jul  8 23:59 __init__.py
-rw-r--r--. 1 root root  5819 Jul  8 23:59 k8s_cluster_info.py
-rw-r--r--. 1 root root  5392 Jul  8 23:59 k8s_exec.py
-rw-r--r--. 1 root root  5576 Jul  8 23:59 k8s_info.py
-rw-r--r--. 1 root root  8264 Jul  8 23:59 k8s_json_patch.py
-rw-r--r--. 1 root root  7418 Jul  8 23:59 k8s_log.py
-rw-r--r--. 1 root root 14009 Jul  8 23:59 k8s.py
-rw-r--r--. 1 root root  6781 Jul  8 23:59 k8s_rollback.py
-rw-r--r--. 1 root root 11323 Jul  8 23:59 k8s_scale.py
-rw-r--r--. 1 root root  7089 Jul  8 23:59 k8s_service.py

```
it is still giving the same error 🤔

**Comment by gavin-stackrox** [comment]
It is probably easier for me to switch the steps to `shell` commands rather than try to resolve the ansible collection issue. Can I get the necessary access to run CI i.e. `/rerun aws`? It'll be quicker than me cobbling together the AWS resources. @mohit-sheth @jtaleric. Thanks.

**Comment by Harshith-umesh** [comment]
18:02:35 TASK [openshift-install : Install dependencies] ********************************
18:02:35 Monday 12 July 2021  12:33:24 +0000 (0:00:03.244)       0:00:16.044 *********** 
18:02:39 [0;31mfatal: [ec2-34-220-206-94.us-west-2.compute.amazonaws.com]: FAILED! => {"changed": false, "failures": ["No package helm available."], "msg": "Failed to install some of the specified packages", "rc": 1, "results": []}[0m

**Comment by gavin-stackrox** [comment]
whats the OS on those EC2 instances?

**Comment by Harshith-umesh** [comment]
fedora 32

**Comment by Harshith-umesh** [comment]
@gavin-stackrox to test out the CI for the time being helm has been installed manually by @mohit-sheth [root@ip-172-31-18-81 ~]# helm version
version.BuildInfo{Version:"v3.4.2", GitCommit:"23dd3af5e19a02d4f4baa5b2f242645a1a3af629", GitTreeState:"clean", GoVersion:"go1.14.13"}
so to test it out , you could remove that line 67 in https://github.com/gavin-stackrox/scale-ci-deploy/blob/82f65f31514e1ccd7ed1ee79a4686810525bdad0/OCP-4.X/roles/openshift-install/tasks/main.yml#L67
also are there any env vars that need to be set to true? coz they will need to be set in the CI files.

**Comment by gavin-stackrox** [comment]
@Harshith-umesh thanks. There are no env vars required (that I know of). I'm firing up a fedora 32 instance in AWS now to see what the pkg name is.

**Comment by Harshith-umesh** [comment]
@gavin-stackrox asked about the env vars coz shouldn't the var RHACS_ENABLE be set to true to test out the changes ?

**Comment by gavin-stackrox** [comment]
> @gavin-stackrox asked about the env vars coz shouldn't the var RHACS_ENABLE be set to true to test out the changes ?

@Harshith-umesh good point.

**Comment by Harshith-umesh** [comment]
updated worker node type to m5.xlarge , since insufficient cpu on workers is causing the ci to fail

**Comment by gavin-stackrox** [comment]
Is this mergeable as is or should I change the RHACS_ENABLE var to false in `CI/properties_files/common_vars.sh`?

**Comment by Harshith-umesh** [comment]
> Is this mergeable as is or should I change the RHACS_ENABLE var to false in `CI/properties_files/common_vars.sh`?

yes , don't think we need to enable rhacs.

**Review by mohit-sheth — APPROVED** [review]
LGTM

**Review by Harshith-umesh — APPROVED** [review]
LGTM, but i wonder if we can incorporate the ansible.core.collections into the playbook itself , or is it necessary that we first install the package manually on a jh before we run the ansible playbook?

**Comment by gavin-stackrox** [comment]
rebased

**Pull request merged**